sherpa/ui/tests/test_session.py

```diff
@@ -1131,12 +1134,29 @@ def test_modelwrapper_init():
     assert repr(wrap) == "<ModelWithDoc model type>"
 
 
+def test_modelwrapper_checks_session():
+    """Check we error out"""
+
+    with pytest.raises(ValueError,
+                       match="session is not a Session instance"):
+        ModelWrapper(ModelWithDoc, None)
+
+
+def test_modelwrapper_checks_model():
+    """Check we error out"""
+
+    s = Session()
+    with pytest.raises(ValueError,
+                       match="modeltype is not a Model instance"):
+        ModelWrapper(s, "Gauss1D")
+
+
 def test_modelwrapper_str_with_doc():
     """Basic ModelWrapper check"""
 
     s = Session()
     wrap = ModelWrapper(s, ModelWithDoc)
-    assert str(wrap) == "This has a doc string"
+    assert str(wrap) == "This has a doc string\n\n    But we don't grab this.\n    "
```

Comment:
Now we do grab it...

Reply:
But we don't - we only grab the "This ha a doc string" part of the message. Or at least that's the intent.

Reply:
Well, not in `test_modelwrapper_what_is_the_docstring_has_doc`, where we don't grab the extra text. In `test_modelwraper_str_with_doc` we do grab it because it's the docstring. I could change

    But we don't grab this.

to something like

    This line is not included in the model-wrapped doc string.

which is probably a bit cleared.

Reply:
I've made this change.